sascha/iban/script.js

```diff
@@ -0,0 +1,176 @@
+        else if(acc.length < 10){
+            //it calculates the difference between 10 and the length of the input and unshifts 0's in at the beginning based on that difference
+            var temp = 10-acc.length;
+            acc = acc.split("");
+				
+            for(var i = 0; i < temp;i++){
+                acc.unshift('0');
+            }
+            acc = acc.join("");
+            return acc;
+            }
+        else if(acc.length > 10){
+            return warningMessage("Ongeldige invoer! Vul a.u.b. een bankrekening onder de 10 cijfers in!");
+        }    
+    }
+   
+    //--------------Getters------------------------------------------------------------------------------------
+    //this function grabs the landcode and injects it into the numbers function
+    var getLandCode = function(){
+        return lettersToNumbers(landCode);
+    }
+    //this function grabs the bankCode which was selected by the user and magically turns it into numbers
+    var getBankCode = function(){
+        return lettersToNumbers(bankCode);
+    }
+    //TODO bankrekeningchecker herschrijven zodat hij eerst dingen check maar wel altijd een value teruggeeft
+    //this function grabs 
+    var getRektnummer = function(){
+        return bankRekeningLengthChecker(bankRekening);    
+    }
+    
+    //gets the whole iban input no, so that the conrol no can be calculated
+    var getWholeIban = function(lcode, bcode, brek){
+        var list = "";
+        
+        list += bcode;
+        list += brek;
+        list += lcode;
+		list += '00';
+        return list;
+    }
+    
+    
+        //deze functie past mod97 toe op bankcode + bankrekening + landcode + 00 en trekt het af van 98 voor het controlegetal
+    function mod97(iban) {
+    var remainder = iban;
+    var checkNo;
+    var block;
+
+        while (remainder.length > 2) {
+            block = remainder.slice(0, 9);
+            remainder = parseInt(block, 10) % 97 + remainder.slice(block.length);
+        }
+        
+        return checkNo = (parseInt(remainder, 10) % 97);
+	}
+	function generateControleGetal(iban){
+        iban = 98- iban; 
+		if(iban < 10){
+            iban = iban.toString();
+            iban = iban.split("");
+            iban.unshift(0);
+            iban = iban.join("");
+        }
+        return iban;
+    }
+    //compiles all calculations and generates the resulting IBAN
+    function executeCalculation(){    
+        var trueIban;
+        bankRekeningLengthChecker(getRektnummer());
+        ibanMathString = getWholeIban(getLandCode(), getBankCode(), getRektnummer());
+        controleGetal = generateControleGetal(mod97(ibanMathString));
+        trueIban = landCode + controleGetal + bankCode + getRektnummer();
+        return trueIban.toString();  
+    }
+    //Validates the IBAN
+    function validateIban(){
+        var check = lettersToNumbers(bankCode) + getRektnummer() + lettersToNumbers(landCode) + controleGetal;

```

Comment:
hier weer het gevaar van JavaScript. de `+`-operator kan zowel getallen optellen, als strings samenvoegen. In dit geval wil je getallen optellen, maar ik heb het idee dat dat niet werkt, omdat getRektnummer() een string teruggeeft. 
(klopt dat?). 

parseInt zou je kunnen helpen.